### PR TITLE
send status for timeout in dmark transport

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

**Comment by rudigiesler** [comment]
Ready for review.

File: vumi/transports/dmark/tests/test_dmark_ussd.py [hunk 1]
```diff
@@ -404,3 +404,23 @@ def test_no_down_status_event_for_bad_responses(self):
...
+        self.tx_helper.dispatch_outbound(msg.reply('foo'))
+        response = yield d
+
+        [status] = yield self.tx_helper.get_dispatched_statuses()
+        self.assertEqual(status['status'], 'down')
+        self.assertTrue(
+            str(self.transport.request_timeout) in status['reasons'][0])
+        self.assertEqual(status['component'], 'response')
+        self.assertEqual(status['type'], 'timeout')
+        self.assertEqual(status['message'], 'Response timed out')
```
Comment (justinvdm) [inline vumi/transports/dmark/tests/test_dmark_ussd.py]: Would it be worth checking the `details` field?

**Comment by justinvdm** [comment]
Left a minor comment, looks good.

**Comment by rudigiesler** [comment]
Changes made. Ready for another review.

**Comment by justinvdm** [comment]
:+1:

**Pull request merged**